squad-server/plugins/fog-of-war.js

```diff
@@ -39,8 +44,17 @@ export default class FogOfWar extends BasePlugin {
   }
 
   async onNewGame() {
-    setTimeout(() => {
-      this.server.rcon.setFogOfWar(this.options.mode);
+    setTimeout(async () => {
+      if (this.options.mode === 1 && this.options.allowedForGameModes.length > 0) {
+        const currentMap = await this.server.rcon.getCurrentMap();
+        this.options.mode = this.options.allowedForGameModes.some((gameMode) =>
```

Comment:
Won't this overwrite the configured value for mode and disable fog of war for all modes once it hits one mode that's not allowed?

Reply:
thanks, good point
 changed to a more secure implementation